feat: Jakarta EE 9 (Vaadin 24 6WP)

MarcinVaadin opened the pull request:
Implementation of: https://github.com/vaadin/flow/issues/14784

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/testbench?pullRequest=1541) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/testbench?pullRequest=1541) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/testbench?pullRequest=1541) it.</sub>

MarcinVaadin commented:
What has been done:
1. Separation of common classes to `vaadin-testbench-shared`
2. Added `vaadin-testbench-core-junit5`
 - Removed parallel scheduler in favor of native JUnit5 parallel support
 - DesiredCapabilities features extracted to new package
 - Multiple capabilities are now run using `TestTemplates` with `InvocationContextProvider`
 - Rules converted to Extensions
3. Added `vaadin-integration-tests-junit5` that are using new approach
 - Parallel runs configured

What needs to be done still:
1. Implement `RetryRule` feature in JUnit5 way

Migration notes:
1. Update imports to `org.junit.jupiter.api.`
2. Annotate test method with `@TestBenchTest` (name to be discussed) instead of `@Test`
3. Extend `AbstractJUnit5TB6Test` instead of `AbstractTB6Test`

Artur- commented on vaadin-testbench-shared/src/main/java/com/vaadin/testbench/TestBench.java:
```diff
@@ -86,7 +87,7 @@ public Object invoke(Object self, Method thisMethod, Method proceed,
     static {
         LicenseChecker.checkLicenseFromStaticBlock("vaadin-testbench",
-                TestBenchVersion.testbenchVersion, null);
+                TestBenchVersion.testbenchVersion, BuildType.DEVELOPMENT);
```
It should be `null` for TestBench to work properly

MarcinVaadin replied:
Reverted, wrong license-checker version (1.7.2) was used in tests

MarcinVaadin commented:
PR splitted into smaller chunks:
- #1547
- #1548
- #1549
- #1550
- #1551
